libraries/wallet/wallet.cpp

```diff
@@ -4182,6 +4292,22 @@ signed_transaction wallet_api::sign_transaction(signed_transaction tx, bool broa
    return my->sign_transaction( tx, broadcast);
 } FC_CAPTURE_AND_RETHROW( (tx) ) }
 
+signed_transaction wallet_api::add_transaction_signature( signed_transaction tx,
```

Comment:
@srpatel19590  - Add the reason for addition of this via comment above this commit.

Reply:
Comment is added in wallet.hpp as below:
Given a fully-formed transaction with or without signatures, signs the transaction with the owned keys and optionally broadcasts the transaction.

This function is called from `cli_multisig_transaction` cli test